typescript/lib/constructor_calls.ts

```diff
@@ -31,12 +36,184 @@ export function newGetFileGenerationOptions(filesToGenerateSet: Set<string>): Ma
+// //                                    Network Context
+// // ====================================================================================================
+
+export function newGetLoadLambdaArgs(lambdaId: LambdaID, image: string, serializedParams: string): LoadLambdaArgs {
+    const result: LoadLambdaArgs = new LoadLambdaArgs();
+    result.setLambdaId(String(lambdaId));
+    result.setContainerImage(image);
+    result.setSerializedParams(serializedParams);
+
+    return result;
+}
+
+export function newGetLambdaInfoArgs(lambdaId: LambdaID): GetLambdaInfoArgs {
+    const result: GetLambdaInfoArgs = new GetLambdaInfoArgs();
+    result.setLambdaId(String(lambdaId)); //TODO - String(lambdaId) VS <string>lambdaId for type assertions (also seen "as")
+
+    return result;
+}
+
+export function newGetRegisterStaticFilesArgs(strSet: Map<string, boolean>): RegisterStaticFilesArgs {
+    const result: RegisterStaticFilesArgs = new RegisterStaticFilesArgs();
+    const staticFilesSetMap: Map<string, boolean> = result.getStaticFilesSetMap();
+    for (let staticFileID in strSet) {
+        staticFilesSetMap.set(staticFileID, strSet[staticFileID]);
+    }
+
+    return result;
+}
+
+export function newGetRegisterFilesArtifactsArgs(filesArtifactIdStrsToUrls: Map<string, string>): RegisterFilesArtifactsArgs {
+    const result: RegisterFilesArtifactsArgs = new RegisterFilesArtifactsArgs();
+    const filesArtifactUrlsMap: Map<string, string> = result.getFilesArtifactUrlsMap();
+    for (let fileArtificactID in filesArtifactUrlsMap) {
+        filesArtifactUrlsMap.set(fileArtificactID, filesArtifactUrlsMap[fileArtificactID]);
+    }
+    return result;
+}
+
+export function newGetRegisterServiceArgs(serviceId: ServiceID, partitionId: PartitionID): RegisterServiceArgs {
+    const result: RegisterServiceArgs = new RegisterServiceArgs();
+    result.setServiceId(<string>serviceId);
+    result.setPartitionId(<string>partitionId);
+
+    return result;
+}
+
+export function newGetStartServiceArgs(
+    serviceId: ServiceID, 
+    dockerImage: string,
+    usedPorts: Map<string, boolean>,
+    entrypointArgs: string[],
+    cmdArgs: string[],
+    dockerEnvVars: Map<string, string>,
+    enclaveDataVolMntDirpath: string,
+    filesArtifactMountDirpaths: Map<string, string>): StartServiceArgs {
+        const result: StartServiceArgs = new StartServiceArgs();
+        result.setServiceId(String(serviceId));
+        result.setDockerImage(dockerImage);
+        const usedPortsMap: Map<string, boolean> = result.getUsedPortsMap();
+        for (let portId in usedPorts) {
+            usedPortsMap.set(portId, usedPorts[portId]);
+        }
+        const entrypointArgsArray: string[] = result.getEntrypointArgsList();
+        for (let entryPoint in entrypointArgs) {
+            entrypointArgsArray.push(entryPoint); //TODO - is this safe, considering arrays are fixed sized
```

Comment:
I _think_ so, though you'll want to double-check it - I think JS will dynamically resize the array for you

Reply:
https://www.xul.fr/javascript/arrays.php
ah yep you're right